bindings/python/iota_sdk/types/block/metadata.py

```diff
@@ -2,66 +2,88 @@
-        shallow_like_parents: Blocks that are directly referenced to adjust opinion.
-        burned_mana: The amount of Mana the Account identified by the IssuerId is at most willing to burn for this block.
-        payload: The optional payload of this block.
+        block_state: The block state.
+        tx_state: The transaction state.
+        block_failure_reason: The block failure reason.
+        tx_failure_reason: The transaction failure reason.
     """
+    block_id: HexStr
+    # TODO: verify if really optional:
+    # https://github.com/iotaledger/tips-draft/pull/24/files#r1293426314
+    block_state: Optional[BlockState]
+    tx_state: Optional[TransactionState]
+    block_failure_reason: Optional[BlockFailureReason]
+    tx_failure_reason: Optional[TransactionFailureReason]
 
-    protocol_version: int
-    strong_parents: List[HexStr]
-    weak_parents: List[HexStr]
-    shallow_like_parents: List[HexStr]
-    burned_mana: str
-    payload: Optional[Union[TaggedDataPayload,
-                      TransactionPayload]] = None
 
-    @classmethod
-    def from_dict(cls, block_dict: Dict) -> Block:
-        """
-        The function `from_dict` takes a dictionary that contains the data needed to
-        create an instance of the `Block` class.
+@json
+@dataclass
+class BlockState(Enum):
+    """Describes the state of a block.
 
-        Returns:
+    Attributes:
+        Pending: Stored but not confirmed.
+        Confirmed: Confirmed with the first level of knowledge.
+        Finalized: Included and can no longer be reverted.
+        Rejected: Rejected by the node, and user should reissue payload if it contains one.
+        Failed: Not successfully issued due to failure reason.
+    """
+    Pending = 0
+    Confirmed = 1
+    Finalized = 2
+    Rejected = 3
+    Failed = 4
 
-        An instance of the `Block` class.
-        """
-        return from_dict(Block, block_dict)
 
-    def id(self) -> HexStr:
-        """Rreturns the block ID as a hexadecimal string.
-        """
-        return Utils.block_id(self)
+@json
+@dataclass
+class TransactionState(Enum):
+    """Describes the state of a transaction.
+
+    Attributes:
+        Pending: Stored but not confirmed.
+        Confirmed: Confirmed with the first level of knowledge.
+        Finalized: Included and can no longer be reverted.
+        Failed: The block is not successfully issued due to failure reason.
+    """
+    Pending = 0
+    Confirmed = 1
+    Finalized = 2
+    Failed = 3
 
 
-class LedgerInclusionState(str, Enum):
-    """Represents whether a block is included in the ledger.
+@json
+@dataclass
+class BlockFailureReason(IntEnum):
+    """Describes the reason of a block failure.
 
     Attributes:
-        noTransaction: The block does not contain a transaction.
-        included: The block contains an included transaction.
-        conflicting: The block contains a conflicting transaction.
+        TooOldToIssue (1): The block is too old to issue.
```

Comment:
(maybe a ticket? could be big)

Reply:
wait a second, what are you referring to exactly? What is the consistency issue? :sweat_smile: My guess would be you refer to the (enum value) in the docs? That's because this is an `IntEnum` where the int value actually has a meaning, while for the `Enum`s they are just necessary syntactically.

Reply:
yea was referring to that, still nice to add for all imo but ok :D